ui/src/components/friends/friends_list/mod.rs

```diff
@@ -336,3 +363,138 @@ pub fn FriendsSkeletal(cx: Scope) -> Element {
+        onclose: move |_| cx.props.did.set(None),
+        show_close_button: false,
+        transparent: false,
+        close_on_click_inside_modal: false,
+        dont_pad: true,
+        div {
+            class: "modal-share-friends",
+            div {
+                padding: "12px",
+                Label {
+                    text: get_local_text("friends.select-chat"),
+                }
+            }
+            state.read().chats_sidebar().iter().filter(|c|cx.props.excluded_chat.map(|id|!c.id.eq(&id)).unwrap_or(true)).cloned().map(|chat| {
+                let participants = state.read().chat_participants(&chat);
+                let other_participants =  state.read().remove_self(&participants);
+                let user: Identity = other_participants.first().cloned().unwrap_or_default();
+                let platform = user.platform().into();
+                // todo: how to tell who is participating in a group chat if the chat has a conversation_name?
+                let participants_name = match chat.conversation_name {
+                    Some(name) => name,
+                    None => State::join_usernames(&other_participants)
+                };
+                let unwrapped_message = match chat.messages.iter().last() {Some(m) => m.inner.clone(),None => raygun::Message::default()};
+                let subtext_val = match unwrapped_message.lines().iter().map(|x| x.trim()).find(|x| !x.is_empty()) {
+                    Some(v) => format_text(v, state.read().ui.should_transform_markdown_text(), state.read().ui.should_transform_ascii_emojis()),
+                    _ => match &unwrapped_message.attachments()[..] {
+                        [] => get_local_text("sidebar.chat-new"),
+                        [ file ] => file.name(),
+                        _ => match participants.iter().find(|p| p.did_key()  == unwrapped_message.sender()).map(|x| x.username()) {
+                            Some(name) => get_local_text_with_args("sidebar.subtext", vec![("user", name)]),
+                            None => {
+                                log::error!("error calculating subtext for sidebar chat");
+                                // Still return default message
+                                get_local_text("sidebar.chat-new")
+                            }
+                        }
+                    }
+                };
+                rsx!(div {
+                    class: "modal-share-friend",
+                    height: "80px",
+                    padding: "16px",
+                    display: "inline-flex",
+                    User {
+                        username: participants_name,
+                        subtext: subtext_val,
+                        timestamp: raygun::Message::default().date(),
+                        active: false,
+                        user_image: cx.render(rsx!(
+                            if chat.conversation_type == ConversationType::Direct {rsx! (
```

Comment:
It would be more idiomatic to use a match condition instead of let. That way if there are any other conversation types, we can expand on it cleanly.